extmod/network_wiznet5k.c

```diff
@@ -686,8 +692,6 @@ STATIC mp_obj_t wiznet5k_make_new(const mp_obj_type_t *type, size_t n_args, size
     #endif
 
     #ifdef MICROPY_HW_WIZNET_SPI_ID
-    // check arguments
-    mp_arg_check_num(n_args, n_kw, 0, 3, false);
     // Allow auto-configuration of SPI if defined for board and no args passed
     if (n_args == 0) {
```

Comment:
maybe also need to add ` && n_kw == 0`, otherwise something like `WIZNET5K(foo=123)` will silently succeed

Reply:
Done. Using a keyword argument raises an error.